pkg/apis/monitoring/v1/types.go

```diff
@@ -316,6 +316,17 @@ type WebConfigFileFields struct {
 	TLSConfig *WebTLSConfig `json:"tlsConfig,omitempty"`
 	// Defines HTTP parameters for web server.
 	HTTPConfig *WebHTTPConfig `json:"httpConfig,omitempty"`
+	// Defines the basic auth users for server.
+	// +optional
+	BasicAuthUsers *BasicAuthUsers `json:"basicAuthUsers,omitempty"`
+}
+
+// BasicAuthUsers defines the basic auth users for prometheus server.
+type BasicAuthUsers struct {
+	// Defines the name of secret stored basic auth users.
+	SecretName *string `json:"secretName"`
```

Comment:
To follow https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#object-references
```suggestion
	  // Defines the secret containing the list of users (keys) with their hashed passwords (values).
	// The secret must contain a key for the service account.
	// +required
	SecretRef SecretReference `json:"secretRef"`
```
with
````golang
type SecretReference struct {
	// Name of the secret in the resource's namespace.
	// +required
	// +kubebuilder:validation:MinLength=1
	Name string `json:"name"`
}
````